src/bundle/Controller/TabController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * File containing the TabController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUiBundle\Controller;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\Core\MVC\Symfony\Controller\Controller;
+use EzSystems\HybridPlatformUi\Response\ResetResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\RouterInterface;
+
+abstract class TabController extends Controller
+{
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var ContentService
+     */
+    private $contentService;
+
+    public function __construct(
+        RouterInterface $router,
+        ContentService $contentService
+    ) {
+        $this->router = $router;
+        $this->contentService = $contentService;
+    }
+
+    protected function resetLocation($contentId)
+    {
+        $mainLocationId = $this->contentService->loadContentInfo($contentId)->mainLocationId;
+
+        return new ResetResponse($this->generateTabUrl($contentId, $mainLocationId, 'full'));
```

Comment:
Maybe a default of `full` could be used for the viewType in `generateTabUrl` as that's what the default is of `_ez_content_view`